velox/exec/Task.cpp

```diff
@@ -653,24 +661,29 @@ void Task::noMoreSplits(const core::PlanNodeId& planNodeId) {
       // Mark all split stores as 'no more splits'.
       for (auto& it : splitsState.groupSplitsStores) {
         it.second.noMoreSplits = true;
-        promises = std::move(it.second.splitPromises);
+        splitPromises = std::move(it.second.splitPromises);
```

Comment:
Hi @mbasmanova , why use assign to instead of append into `splitPromises` here?

Reply:
@waitinfuture Are you saying that we may have multiple entries in splitsState.groupSplitsStores, but we keep only one of these in splitPromises. Sounds like a bug. 

@spershin Sergey, would you like to take a look?

Reply:
@waitinfuture , @mbasmanova 
This is not a bug. As comment on lines 655-657 says 'noMoreSplits' comes only for ungrouped execution when we only have a single group.

The code is a bit confusing though and it looks like if might have been initially written with idea that 'noMoreSplits' could come for grouped execution too - thus the loop over the 'groupSplitsStores'.
In that case it would have been a bug which we would like have found already.

So, I'll go ahead and modify this code to remove the confusion.

Thanks for point it out.

Curious, @waitinfuture , how did it happen that you paid attention on this? Is there an issue you've been investigating?

Reply:
@waitinfuture , @mbasmanova
Actually, re-reading the comment again, it seems like this message can come for a grouped execution plan node.
It will mean 'no more split groups' for this node.
So, this is a bug indeed.
Thanks for pointing out!

Reply:
https://github.com/facebookincubator/velox/pull/8897 to fix this.

Reply:
Thanks @spershin ! I was reading the code and got confused. As you said, to me the comment means both ungrouped and grouped execution can happen. I have limited knowledge for the contract how messages should be passed (i.e. 'no more splits for group' should be acknowledged before 'no more split groups' is sent), and I think the fix is good for integrating velox into other engines.